### ISSUE TITLE
[Java] false negative: jdo-sqli misses `"...".formatted(input)` pattern.

### ISSUE THREAD

**Issue by 9iang22** [returntocorp/sgrep-rules#3812]
## Describe the bug  
The jdo-sqli rule misses dynamic query construction that uses Java’s instance formatting API String.formatted(...).
The rule currently detects String.format(...) and concatenation patterns, but does not flag equivalent "...%s...".formatted(input) usage when passed to JDO query APIs. 

This leads to false negatives for potentially user-controlled query construction.
Rule: java/lang/security/audit/sqli/jdo-sqli.yaml  

### To Reproduce  
```java
package testcode.sqli;
import javax.jdo.PersistenceManager;
import java.util.ArrayList;
public class JdoSql {
    public void testJdoQueriesAdditionalMethodSig(String input) {
        PersistenceManager pm = getPM();
        // ruleid: jdo-sqli
        pm.newQuery(UserEntity.class, new ArrayList(), "id == %s".formatted(input));
    }
}
```

For comparison, this variant is detected:
```java
pm.newQuery(UserEntity.class, new ArrayList(), String.format("id == %s", input));
```

## Priority   
- [ ] P0: blocking me from making progress  
- [ ] P1: this will block me in the near future  
- [x] P2: annoying but not blocking me
